src/main/resources/avro/bdg.avdl

```diff
@@ -257,32 +257,43 @@ record AlignmentRecord {
   union { null, Contig } mateContig = null;
 }
 
+enum SequencingPlatform {
+  Illumina
+}
+
+record IlluminaInfo {
+  union { null, string } flowcellId;
+  union { null, int } lane;
+  union { null, int } tile;
+  union { null, int } surface;
+  union { null, int } xPosition;
+  union { null, int } yPosition;
+}
+
 record Sequence {
+  /**
+   Analogous to fastq
+   */
+  union { null, string } sequenceName = null;
   union { null, string } bases = null;
   union { null, string } qualities = null;
+  union { null, SequencingPlatform } platformType = null;
+  union { null, IlluminaInfo } platformInfo = null;
 }
 
 /**
-   The DNA fragment that is was targeted by the sequencer, resulting in
-   one or more reads.
+   A set of objects that should be physically analyzed/stored together.
+   Typically the result of some type of groupby operation.
```

Comment:
two nits on this comment:

1. "A set of objects" feels vague bordering on cryptic. They're always "reads", right?
2. I'm confused by the idea that you'd get these as a result of a groupby; I assume we're all talking about taking an `RDD[AlignmentRecord]` and doing a `.map(r => r.getReadName -> r).groupByKey` to it… but then where does `Bucket.sequences` come from / get populated? Is the answer just that it's a little more complicated than that? i.e.

```
def groupReadsIntoBuckets(rdd: RDD[AlignmentRecord]): RDD[Bucket] =
  rdd
    .map(r => r.getReadName -> r)
    .groupByKey
    .mapValues(reads => 
      Bucket
        .newBuilder
        .setBucketName(reads.head.getReadName)
        .setSequences(reads.map(_.getSequence))  // is this the right field?
        .setAlignments(reads)
        .build
    )
```